From 4a67893f1d36fc2611a63eaac30eea3b9e69213d Mon Sep 17 00:00:00 2001
From: Asahi Lina <lina@asahilina.net>
Date: Wed, 13 Nov 2024 03:47:03 +0900
Subject: [PATCH] FEXLoader: Align stack base

This ensures that __libc_stack_end is aligned, the same way it is on
native.
---
 Source/Tools/FEXLoader/ELFCodeLoader.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Source/Tools/FEXLoader/ELFCodeLoader.h b/Source/Tools/FEXLoader/ELFCodeLoader.h
index 8e24f75b83..733be0317f 100644
--- a/Source/Tools/FEXLoader/ELFCodeLoader.h
+++ b/Source/Tools/FEXLoader/ELFCodeLoader.h
@@ -725,6 +725,9 @@ class ELFCodeLoader final : public FEX::CodeLoader {
     uint64_t ExecFNLocation = TotalArgumentMemSize;
     TotalArgumentMemSize += Args[0].size() + 1;
 
+    // Align the argument block to 16 bytes to keep the stack aligned
+    TotalArgumentMemSize = FEXCore::AlignUp(TotalArgumentMemSize, 16);
+
     // Offset the stack by how much memory we need
     StackPointer -= TotalArgumentMemSize;
 
